### DISCUSSION TITLE
Unload combine - question about unloading trailer

### DISCUSSION THREAD

**Discussion by bojanh66** [Courseplay/Courseplay_FS22#1955] [Q&A] [answered]
Hello, i am using newest CP version and unloading combine works perfect.
But, once the trailer is full, what should i do now?
In CP description, it says :
`With a trailer, Courseplay will stop when full and you have to set up AutoDrive to bring the tractor to a spot where it can empty the trailer. Alternatively, click that little steering wheel in the HUD and toggle through a list of targets to let the Giants helper handle the transport for you. Note that we did not improve the Giants helper...`

When i click on wheel icon, it will use only Giants helper, which i dont want, i want to use Autodrive.

Can someone explain me how to set up Autodrive to unload full trailer, which mode should i use, do i need to tick some option?
I have selected in target my Yard (farm) Silo, but it doesnt work.
Help please.
![FarmingSimulator2022Game 2022-09-30 08-14-55](https://user-images.githubusercontent.com/32069662/193203024-4d74c020-d328-42d2-8370-0988981a0f92.png)

**Comment by Tensuko** [comment, marked as answer]
If you want to use AD then you got to set it up and I think it just works like for example a loading wagon.

The mode you selected in AD is only drive to target. For more AD questions pls check AD github Wiki Page or AD Discord.

**Reply by rohne83** [reply]
yes, needs to be set up same as loading wagon: mode 'pickup&deliver', CP button in AD hud activated, and of course AD 'drive on' fill level set to a low enough value (has to be lower than the one set in CP, otherwise AD will directly hand back to CP as its not full enough).
But that's it so far.

**Comment by TinMen6** [comment]
what i liked when using cp in the old fs games was the 2 points system, so 1 point where the tractor will wait until the combine lvl is reached and a 2 point where the tractor will leave the field. It was super handy when using more than one tractor to unload the combine. Do you guys plan to get this option in this version of cp?

**Reply by Tensuko** [reply]
That was when CP drives to unload, but that wont come back.
CP waits where you start it.

**Reply by TinMen6** [reply]
That is too bad imo it was a great system. How does the AutoRefill and Auger unloading works can't find any info about that.

**Reply by Tensuko** [reply]
Have a look at AutoDrive for transport and refill and stuff.
Auger Wagon check the changelog of 7.1.1.0

**Reply by TinMen6** [reply]
Thanks for all the help, i figured out the fertilizer refill and i will check the log a bit latter. Have a nice day.

**Comment by Lactic68** [comment]
Hi @Tensuko ,
I have another question about this Combine Unload mode. Is there a way to have multiple auger wagon on hold when watching combine signals to unload ? In bigger fields it might be very handy to have multiple auger wagon controlled by CP to unload combine.

**Reply by Tensuko** [reply]
Sure can.
It even is controlled by score about distance and fill level. But focus is to make one auger full first.
So it can start unload and be back asap.

**Comment by PhilippS1** [comment]
Hallo,
schön wäre es auch wenn der Überladewagen auch die Möglichkeit hätte mit dem AD weiterfahren zu können.

**Reply by Tensuko** [reply]
Zu welchem Zweck?
Wir haben das extra simpel gehalten.
Ich sehe keinen Sinn mit einem ÜLW die Ernte über Straßen zu transportieren.

**Comment by PhilippS1** [comment]
Natürlich hast du recht das es nicht unbedingt Sinn macht mit einem ÜLW längere Strecken auf der Straße zu transportieren. Es macht aber auch nicht immer Sinn mit dem Abfahrer für den ÜLW direkt bis ans Feld zu fahren noch dazu wenn der Sattelauflieger sich dann im beladenen Zustand beim Verlassen des Feldes fest fährt. Außerdem wenn das Feld zu groß wird findet der ÜLW den Abfahrer nicht mehr.

Woran kann es liegen das der ÜLW den Abfahrer zwar richtig erkennt jedoch auf der falschen Seite anfährt?
Mod-Anhänger
![fsScreen_2022_12_29_17_45_29](https://user-images.githubusercontent.com/55155040/209984226-2075c409-5923-464d-8e2c-2fa05d14590f.png)
Strandard-Anhänger

![fsScreen_2022_12_29_17_50_19](https://user-images.githubusercontent.com/55155040/209984277-e2ff3e61-fb7a-4b18-bdfa-0dad55998972.png)

**Reply by Tensuko** [reply]
> Natürlich hast du recht das es nicht unbedingt Sinn macht mit einem ÜLW längere Strecken auf der Straße zu transportieren. Es macht aber auch nicht immer Sinn mit dem Abfahrer für den ÜLW direkt bis ans Feld zu fahren noch dazu wenn der Sattelauflieger sich dann im beladenen Zustand beim Verlassen des Feldes fest fährt. Außerdem wenn das Feld zu groß wird findet der ÜLW den Abfahrer nicht mehr.
> 
> Woran kann es liegen das der ÜLW den Abfahrer zwar richtig erkennt jedoch auf der falschen Seite anfährt? Mod-Anhänger ![fsScreen_2022_12_29_17_45_29](https://user-images.githubusercontent.com/55155040/209984226-2075c409-5923-464d-8e2c-2fa05d14590f.png) Strandard-Anhänger
> 
> ![fsScreen_2022_12_29_17_50_19](https://user-images.githubusercontent.com/55155040/209984277-e2ff3e61-fb7a-4b18-bdfa-0dad55998972.png)

Falsche Seite -> Mod Problem. Da hatten wir auch ein Issue zu irgendwo. Wurde bei einem Trailer sogar korrigiert.
Findet Abfahrer nicht mehr bei zu großem Feld glaub ich erst mal nicht. Da hilft nur richtige Infos um was dazu sagen zu können.
Wenn das so für dich nicht passt, dann nutzt die die dir gegebenen möglichkeiten nicht richtig.
ÜLW sind genau dafür gemacht, auf dem Feld Drescher oder Ernter ab zu fahren und in Trailer am Feldrand über zu laden.
Sonst muss du direkt Trailer benutzen und auf der Straße fahren lassen.

**Reply by PhilippS1** [reply]
Weißt du noch welches Issue es war? 
Gibt es denn einen Timeout bei der Suche nach einem Abfahrer?
Die Frage nach einer Möglichkeit zur Übergabe an AD wäre für mein Problem die einfachste Lösung.

**Reply by Tensuko** [reply]
Issue: #2067

Nach ein paar Versuchen wird aufgegeben, das heist aber nicht, dass das Ziel nicht gefunden wurde, eher Probleme mit der Pfadgenerierung und das kann diverse Ursachen haben.

Das mag sein, aber ist nicht in unserem Sinn.

**Reply by rohne83** [reply]
Wobei ich die Intention durchaus nachvollziehen kann. Im 19er war meine Standardmap die Mittelsachsen, und da gab es tatsächlich das ein oder andere Feld wo partout kein vernünftiger Wartepunkt für die Abfahrer möglich war, und ich entsprechend dann weit ab vom Feld überladen lassen musste. Solche Abfahrer würde CP ja nicht finden, da zu weit weg vom Feldrand. An AD übergeben wäre in solchen Fällen dann doch auch für den ULW mal hilfreich.
Relativ spezieller Fall, und auch nicht derart bedeutend dass man jetzt die Logik dafür ändern sollte, aber zumindest nicht vollständig aus der Luft gegriffen ;)

**Comment by PhilippS1** [comment]
Passt danke für deine Hilfe.
